src/NuGet.Core/NuGet.ProjectModel/CentralTransitiveDependencyGroup.cs

```diff
@@ -28,7 +28,7 @@ public CentralTransitiveDependencyGroup(NuGetFramework framework, IEnumerable<Li
 
         public string FrameworkName { get; }
 
-        public IEnumerable<LibraryDependency> TransitiveDependencies { get; }
+        public IList<LibraryDependency> TransitiveDependencies { get; }
```

Comment:
One place I see is that PackTaskLogic.AddDependecies takes this in as an IList, and updates the list in place (rather than mutating the LibraryDependency directly)

Reply:
I think that's a bug in the pack code. 
Pack should be appending the central transitive groups, not editing them in place.

Reply:
> > The reason why I'm asking is because a foreach would usually allocate on an IList
> 
> Maybe we should use for `for` loops and fewer `foreach` loops 😄

We can. 

The other thing about this type is that I'm really worried about a few allocations in the pack codepaths.

Reply:
To clarify, you are worried about allocations introduced in this PR, or allocations in general?

Reply:
My brain though no, my fingers didn't type no. 

I'm really *not* worried about pack codepath allocations, which seem to be the ones to benefit from this change.